aspnetcore/fundamentals/environments.md

```diff
@@ -1,601 +1,580 @@
+To determine the runtime environment, ASP.NET Core reads from the following environment variables:
 
-The following code:
+:::moniker range=">= aspnetcore-7.0"
 
-* Is similar to the code generated by the ASP.NET Core templates.
-* Enables the [Developer Exception Page](xref:fundamentals/error-handling#developer-exception-page) when `ASPNETCORE_ENVIRONMENT` is set to `Development`. This is done automatically by the <xref:Microsoft.AspNetCore.Builder.WebApplication.CreateBuilder%2A?displayProperty=nameWithType> method.
-* Calls <xref:Microsoft.AspNetCore.Builder.ExceptionHandlerExtensions.UseExceptionHandler%2A> when the value of `ASPNETCORE_ENVIRONMENT` is anything other than `Development`.
-* Provides an <xref:Microsoft.AspNetCore.Hosting.IWebHostEnvironment> instance in the <xref:Microsoft.AspNetCore.Builder.WebApplication.Environment> property of `WebApplication`.
+* [`DOTNET_ENVIRONMENT`](xref:fundamentals/configuration/index#default-host-configuration)
+* `ASPNETCORE_ENVIRONMENT` when the <xref:Microsoft.AspNetCore.Builder.WebApplication.CreateBuilder%2A?displayProperty=nameWithType> method is called. The ASP.NET Core web app project templates call `WebApplication.CreateBuilder`. The `DOTNET_ENVIRONMENT` value overrides `ASPNETCORE_ENVIRONMENT` when <xref:Microsoft.AspNetCore.Builder.WebApplicationBuilder> is used. For other hosts, such as <xref:Microsoft.Extensions.Hosting.GenericHostBuilderExtensions.ConfigureWebHostDefaults%2A> and <xref:Microsoft.AspNetCore.WebHost.CreateDefaultBuilder%2A?displayProperty=nameWithType>, `ASPNETCORE_ENVIRONMENT` has higher precedence.
 
-:::code language="csharp" source="environments/samples/6.x/EnvironmentsSample/Program.cs" id="First":::
+:::moniker-end
 
-The [Environment Tag Helper](xref:mvc/views/tag-helpers/builtin-th/environment-tag-helper) uses the value of [IHostEnvironment.EnvironmentName](xref:Microsoft.Extensions.Hosting.IHostEnvironment.EnvironmentName) to include or exclude markup in the element:
+:::moniker range=">= aspnetcore-6.0 < aspnetcore-7.0"
 
-:::code language="cshtml" source="environments/samples/6.x/EnvironmentsSample/Pages/About.cshtml" id="snippet_Environment":::
+* [`DOTNET_ENVIRONMENT`](xref:fundamentals/configuration/index#default-host-configuration)
+* `ASPNETCORE_ENVIRONMENT` when the <xref:Microsoft.AspNetCore.Builder.WebApplication.CreateBuilder%2A?displayProperty=nameWithType> method is called. The ASP.NET Core web app project templates call `WebApplication.CreateBuilder`. The `ASPNETCORE_ENVIRONMENT` value overrides `DOTNET_ENVIRONMENT`.
 
-The [About page](https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/fundamentals/environments/samples/6.x/EnvironmentsSample/Pages/About.cshtml) from the [sample code](https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/fundamentals/environments/samples/6.x) includes the preceding markup and displays the value of `IWebHostEnvironment.EnvironmentName`.
+:::moniker-end
```

Comment:
Isn't this saying pretty much the same thing as the >= aspnetcore-7.0 version above?

Reply:
The override is reversed (at least it says so, as I never kept this tidbit in my head) ...

* .NET 7: `DOTNET_ENVIRONMENT` value overrides `ASPNETCORE_ENVIRONMENT`
* .NET 6: `ASPNETCORE_ENVIRONMENT` value overrides `DOTNET_ENVIRONMENT`

Reply:
Ah, I need to learn to read more carefully! Thanks!